api/v1alpha4/common_types.go

```diff
@@ -31,6 +31,24 @@ const (
 	// tool uses this label for implementing provider's lifecycle operations.
 	ProviderLabelName = "cluster.x-k8s.io/provider"
 
+	// ClusterNameAnnotation is the annotation set on nodes identifying the name of the cluster the node belongs to.
+	ClusterNameAnnotation = "cluster.x-k8s.io/cluster-name"
+
+	// ClusterNamespaceAnnotation is the annotation set on nodes identifying the namespace of the cluster the node belongs to.
+	ClusterNamespaceAnnotation = "cluster.x-k8s.io/cluster-namespace"
+
+	// MachineAnnotation is the annotation set on nodes identifying the machine the node belongs to.
+	MachineAnnotation = "cluster.x-k8s.io/machine"
+
+	// MachinePoolAnnotation is the annotation set on nodes identifying the machine pool the node belongs to.
+	MachinePoolAnnotation = "cluster.x-k8s.io/machine-pool"
```

Comment:
I think that in case of MachinePools we should not add a different annotation but instead add something like

```
cluster.x-k8s.io/owner-kind: MachinePool
cluster.x-k8s.io/owner-name: capz-cluster-2-mp-0
```

So all the nodes managed by higher-level abstraction have a consistent set of labels.
The only exception for the machine pools is the `cluster.x-k8s.io/machine` being missing

Reply:
Given that these are annotations, could we instead json-marshal the owner reference as we see it today instead of creating a new standard to store this information?

Reply:
what do you mean by json-marshal the owner reference? How would it be represented in the annotations? Would it be one annotation for the entire owner ref?

Reply:
Something like `cluster.x-k8s.io/owner` and its value would be the json value of an OwnerReference object

Reply:
IMO 

```
cluster.x-k8s.io/owner-kind: MachinePool
cluster.x-k8s.io/owner-name: capz-cluster-2-mp-0
``` 

is more readable than dumping the json string in there. I don't really care either way, what I really wanted from this PR in the first place was the cluster name, owner name + kind were just added on for additional info.

Reply:
Sounds good to me, I assume that would depend on who's using these and how, for example if this is only for human users that's probably fine. For machines / controllers the information above for example is missing the API group (this is only a problem if we ever say that a Machine could be controlled outside of Cluster API main types)